### PR TITLE
docs: add dev guide overview

### PR DESCRIPTION
**Pull request by Benjozork**

### CODE REVIEW

File: docs/a32nx-dev/overview.md [hunk 1]
```diff
@@ -0,0 +1,116 @@
+# FlyByWire Development Guide Overview
+
+## Introduction
+
+This guide has the objective of teaching you how to contribute efficiently to the FlyByWire A32MX project.
```
Comment (pepperoni505) [inline docs/a32nx-dev/overview.md]:
```suggestion
This guide has the objective of teaching you how to contribute efficiently to the FlyByWire A32NX project.
```

File: docs/a32nx-dev/overview.md [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+
+You will learn how to:
+
+- Setup a functional development environment;
+- Get familiar with the technologies involved;
+- Understand our build process;
+- Integrate into our development cycle and best practices.
```
Comment (pepperoni505) [inline docs/a32nx-dev/overview.md]: No need for a semicolon at the end of the line and no need for the period on the last item
```suggestion
- Setup a functional development environment
- Get familiar with the technologies involved
- Understand our build process
- Integrate into our development cycle and best practices
```

File: docs/a32nx-dev/overview.md [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+
+## Setting up a development environment
+
+Working on the addon will require, at a minimum, the following software:
```
Comment (pepperoni505) [inline docs/a32nx-dev/overview.md]:
```suggestion
Working on the addon will require at a minimum the following software:
```

File: docs/a32nx-dev/overview.md [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+
+- A code editor (with at least syntax highlighting, search and replace, and support for workspaces)
+- A `git` client (along with any editor integrations of your choice)
+- Docker desktop (either WSL2 or HyperV backend work, but the latter is faster)
```
Comment (pepperoni505) [inline docs/a32nx-dev/overview.md]:
```suggestion
- Docker Desktop (either WSL2 or HyperV backend work, but the latter is faster)
```
the official name is Docker Desktop, so should be capitalized

File: docs/a32nx-dev/overview.md [hunk 5]
```diff
@@ -0,0 +1,116 @@
...
+
+Most of our team works with either Visual Studio Code or IntelliJ IDEA-based IDEs for development. Obviously, your choice is yours as long as the resulting code conforms to our standards.
+
+## Used technologies
+
+Our tech stack includes the following:
+
+- Systems development for aircraft uses Rust and the `msfs-rs` library
```
Comment (pepperoni505) [inline docs/a32nx-dev/overview.md]:
```suggestion
- Systems development for aircraft using Rust and the `msfs-rs` library
```

File: docs/a32nx-dev/overview.md [hunk 6]
```diff
@@ -0,0 +1,116 @@
...
+- Avionics programming is done using JavaScript or TypeScript (depending on the project), with the `React.js` rendering library.
+- Front-end web or desktop app development uses the same technologies outlined above
+- Server-side development uses `nest.js` for the API and `MySQL` for the database backend
+
+Knowledge of all items on this list is obviously not necessary, but this can hopefully give you a better insight into what your skills can fit into.
+
+## Build process
+
+Due to how Microsoft Flight Simulator package building works, it is necessary for us to maintain our own, bespoke build tools.
```
Comment (pepperoni505) [inline docs/a32nx-dev/overview.md]:
```suggestion
Due to how Microsoft Flight Simulator package building works, it is necessary for us to maintain our own bespoke build tools.
```

File: docs/a32nx-dev/overview.md [hunk 7]
```diff
@@ -0,0 +1,116 @@
...
+
+The high-level objective involving the build process is to eventually have all code in the `src/` directory.
+
+Commonly, the following items will be found in the source:
```
Comment (pepperoni505) [inline docs/a32nx-dev/overview.md:41]: Maybe rephrase this?

File: docs/a32nx-dev/overview.md [hunk 8]
```diff
@@ -0,0 +1,116 @@
...
+- src/
+    - instruments/                  Avionics code built using the rollup config
+        - rollup.config.js
+    - behavior/                     Compiled and verified ModelBehavior code
+    - model/                        Model source files to be merged at build time
+    - <directories for systems>     Rust/C++ source code for systems
+    ... more to come
+```
+
+Each directory will usually contain a build.js/build.sh file that is called from a central build script located in `srcipts/build.sh`.
```
Comment (pepperoni505) [inline docs/a32nx-dev/overview.md]:
```suggestion
Each directory will usually contain a build.js/build.sh file that is called from a central build script located in `scripts/build.sh`.
```

File: docs/a32nx-dev/overview.md [hunk 9]
```diff
@@ -0,0 +1,116 @@
...
+
+A Docker container is used to ship pre-configured build environments to developers. It contains necessary library headers and compilers, and requires no user intervention at all.
+
+This build process is optimized to be run in a cloud CI pipelines on GitHub Actions. This allows us to automatically distribute development master builds to thousands of users without any manual intervention. It lets the team deploy critical fixes to production in a matter of minutes.
```
Comment (pepperoni505) [inline docs/a32nx-dev/overview.md]:
```suggestion
This build process is optimized to be run in cloud CI pipelines on GitHub Actions. This allows us to automatically distribute development master builds to thousands of users without any manual intervention. It lets the team deploy critical fixes to production in a matter of minutes.
```

File: docs/a32nx-dev/overview.md [hunk 10]
```diff
@@ -0,0 +1,116 @@
...
+
+FlyByWire Simulations aims to maintain professional standards as well as industry-wide best practices in order to maximize developer productivity and deliver high-quality, tested products.
+
+This is why we adopt a set of rules and practices that create a strong framework, not only around the usage of `git` but also surrounding code quality.
+
+### Version control
+
+Our version control approach maintains the following guidelines:
+
+- Commit messages should be short and to the point. The Angular commit message convention is often used and recommended but not enforced at the moment
```
Comment (pepperoni505) [inline docs/a32nx-dev/overview.md]: Should probably link to the Angular commit message convention

File: docs/a32nx-dev/overview.md [hunk 11]
```diff
@@ -0,0 +1,116 @@
...
+
+Code quality is of an utmost importance as it allows other developers to more easily find bugs and understand logic.
+
+An ESLint configuration is provided and should cover most code style rules.
+
+A few guidelines apply:
+
+- Prefer self-documenting code over having to write tons of comments
+- However, do not hesitate to write comments when complex logic is used
+- Use language-specific docstring tools (JSDoc, rustdoc)
```
Comment (pepperoni505) [inline docs/a32nx-dev/overview.md:90]: maybe link to these?

File: docs/a32nx-dev/overview.md [hunk 12]
```diff
@@ -0,0 +1,116 @@
...
+- Maintain separation of concerns
+- Do not be afraid to separate code into logical blocks, even if resulting sub-functions are only used locally (while maintaining a reasonable limit)
+- Avoid polling - prefer reactive APIs (but remain sure of not introducing state that cannot be externally read)
```
Comment (pepperoni505) [inline docs/a32nx-dev/overview.md:93]:
```suggestion
- Avoid polling - prefer reactive APIs (but make sure you're not introducing states that cannot be externally read)
```

File: docs/a32nx-dev/overview.md [hunk 13]
```diff
@@ -0,0 +1,116 @@
...
+- Avoid introducing new technologies/languages without talking with the development team
+
+Additionally, a few practices are straight up forbidden:
+
+- Do not use `async/await` without a good reason (it does not make code run faster unless you are calling threaded APIs like `fetch`)
+- Do not use hungarian notation (b, m, g identifier prefixes)
+- Do not perform business logic calculations in presentation code
```
Comment (pepperoni505) [inline docs/a32nx-dev/overview.md:100]: I don't quite understand this line (line 100)

File: docs/a32nx-dev/overview.md [hunk 14]
```diff
@@ -0,0 +1,116 @@
...
+
+### Reference material
+
+Creating a high-fidelity aircraft is a serious task that requires focus and dedication. But more importantly, it's a strive for accuracy and realism. Therefore, any change that affects the appearance or behavior of a plane needs to reference IRL documentation.
```
Comment (pepperoni505) [inline docs/a32nx-dev/overview.md]:
```suggestion
Creating a high-fidelity aircraft is a serious task that requires focus and dedication. But more importantly, it's a strive for accuracy and realism. Therefore, any change that affects the appearance or behavior of a plane needs to reference real documentation.
```
IRL doesn't sound professional

File: docs/a32nx-dev/overview.md (hunk 4 above)
Comment (pepperoni505) [inline docs/a32nx-dev/overview.md]: Link to some of these

File: docs/a32nx-dev/overview.md [hunk 15]
```diff
@@ -0,0 +1,116 @@
...
+Most of our team works with either Visual Studio Code or IntelliJ IDEA-based IDEs for development. Obviously, your choice is yours as long as the resulting code conforms to our standards.
+
+## Used technologies
+
+Our tech stack includes the following:
+
+- Systems development for aircraft uses Rust and the `msfs-rs` library
+- Avionics programming is done using JavaScript or TypeScript (depending on the project), with the `React.js` rendering library.
+- Front-end web or desktop app development uses the same technologies outlined above
+- Server-side development uses `nest.js` for the API and `MySQL` for the database backend
```
Comment (pepperoni505) [inline docs/a32nx-dev/overview.md]: Again, should link to some things

File: docs/a32nx-dev/overview.md (hunk 12 above)
Comment (Benjozork) [reply docs/a32nx-dev/overview.md:93]: state should stay singular as we're talking about state in general

**Comment by Valastiri** [comment]
As a heads up, mkdocs-material doesn't drop header dividers on it's own. You'd have to specify on it's own line `---` underneath sections. 

image

** Forgot to add sample image

image

If you wanted to add dividers or just one under the main heading. (which probably could be a standard we could do just for formatting)

But it looks good yea.

**Review by pepperoni505 — APPROVED** [review]
LGTM
